common4j/src/main/com/microsoft/identity/common/java/commands/parameters/nativeauth/BaseSignUpStartCommandParameters.java

```diff
@@ -0,0 +1,54 @@
+//  Copyright (c) Microsoft Corporation.
+//  All rights reserved.
+//
+//  This code is licensed under the MIT License.
+//
+//  Permission is hereby granted, free of charge, to any person obtaining a copy
+//  of this software and associated documentation files(the "Software"), to deal
+//  in the Software without restriction, including without limitation the rights
+//  to use, copy, modify, merge, publish, distribute, sublicense, and / or sell
+//  copies of the Software, and to permit persons to whom the Software is
+//  furnished to do so, subject to the following conditions :
+//
+//  The above copyright notice and this permission notice shall be included in
+//  all copies or substantial portions of the Software.
+//
+//  THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+//  IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+//  FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+//  AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+//  LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+//  OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+//  THE SOFTWARE.
+package com.microsoft.identity.common.java.commands.parameters.nativeauth;
+
+import java.util.Map;
+
+import edu.umd.cs.findbugs.annotations.Nullable;
+import lombok.EqualsAndHashCode;
+import lombok.Getter;
+import lombok.NonNull;
+import lombok.experimental.SuperBuilder;
+
+/**
+ * BaseSignUpTokenCommandParameters is the base class for parameters for all all Native Auth sign up related commands.
+ */
+@Getter
+@EqualsAndHashCode(callSuper = true)
+@SuperBuilder(toBuilder = true)
+public class BaseSignUpStartCommandParameters extends BaseNativeAuthCommandParameters {
+    private static final String TAG = BaseSignUpStartCommandParameters.class.getSimpleName();
+
+    /**
+     * The email address of the user.
+     */
+    @NonNull
+    public final String username;
+
+    /**
+     * The user attributes of the user set in the user flow need to be collected.
```

Comment:
Are these (all of these) recorded on the user object in the directory?

Also are these attributes essentially claims in the eventual tokens?

Reply:
Yes, these attributes are part of token claim.